File: libraries/CmpCoinbase/composeBase/src/commonMain/kotlin/dev/subfly/cmpcoinbase/core/data/dto/CoinDto.kt
package dev.subfly.cmpcoinbase.core.data.dto

import kotlinx.serialization.Serializable

@Serializable
data class CoinDto(
    val id: String? = null,
    val contract: String? = null,
    val contracts: List<ContractDto>? = null,
    val description: String? = null,
    val development_status: String? = null,
    val first_data_at: String? = null,
    val hardware_wallet: Boolean? = null,
    val hash_algorithm: String? = null,
    val is_active: Boolean? = null,
    val is_new: Boolean? = null,
    val last_data_at: String? = null,
    val links: LinksDto? = null,
    val links_extended: List<LinksExtendedDto>? = null,
    val logo: String? = null,
    val message: String? = null,
    val name: String? = null,
    val open_source: Boolean? = null,
    val org_structure: String? = null,
    val parent: ParentDto? = null,
    val platform: String? = null,
    val proof_type: String? = null,
    val rank: Int? = null,
    val started_at: String? = null,
    val symbol: String? = null,
    val tags: List<TagDto>? = null,
    val team: List<TeamDto>? = null,
    val type: String? = null,
    val whitepaper: WhitePaperDto? = null
)

File: ios/BaseiOSSwitUI/BaseiOSSwitUI/Core/Data/Dto/CoinDto.swift
//
//  CoinDto.swift
//  BaseiOSSwitUI
//
//  Created by Ali Taha on 4.12.2023.
//

import Foundation

struct CoinDto: Codable, Identifiable {
    let id: String?
    let contract: String?
    let contracts: [ContractDto]?
    let description: String?
    let development_status: String?
    let first_data_at: String?
    let hardware_wallet: Bool?
    let hash_algorithm: String?
    let is_active: Bool?
    let is_new: Bool?
    let last_data_at: String?
    let links: LinksDto?
    let links_extended: [LinksExtendedDto]?
    let logo: String?
    let message: String?
    let name: String?
    let open_source: Bool?
    let org_structure: String?
    let parent: ParentDto?
    let platform: String?
    let proof_type: String?
    let rank: Int?
    let started_at: String?
    let symbol: String?
    let tags: [TagDto]?
    let team: [TeamDto]?
    let type: String?
    let whitepaper: WhitePaperDto?
}
